.github/workflows/python-publish.yml

```diff
@@ -57,8 +57,39 @@ jobs:
         with:
           path: dist/*.tar.gz
 
-  upload_pypi:
+  test_wheels:
     needs: [build_wheels, build_sdist]
+    name: Run tests on wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 2
+      matrix:
+        os: [ubuntu-18.04]
+    steps:
+      - uses: actions/download-artifact@v2
+        name: Download artifacts
+        with:
+          name: artifact
+          path: dist
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.6'
+
+      - name: Install wheel and tests dependencies
+        run: |
+          pip install $(echo dist/*manylinux2010_x86_64.whl)[tests]
+
+      - name: Run tests
+        run: |
+          xdoctest pyflann --style=google all && \
+          tar xf dist/*.tar.gz --wildcards wbia-pyflann*/test/ && \
+          cd wbia-pyflann*/test && \
+          pytest
```

Comment:
What do you think about encapsulating these commands in a script?

I'm suggesting that so that this file can continue to copy&paste this file throughout the projects without minor modifications. So possibly having this step call `scripts/test.sh` as a layer of abstraction. Then we can do whatever needs to be different in that script file. Thoughts?

Reply:
Where can this file live so it can be accessed in the github workflow...?

Reply:
Ah, I see the issue in that case. When I made the comment I was thinking about checkout code.
Generally speaking this entire step is dependent on knowing about the project since it needs the name in several places. Maybe this isn't the best approach after all.

Reply:
I would suggest using a script, as Michael suggests, with the `CIBW_TEST_COMMAND` as employed by vtool: https://github.com/WildbookOrg/wbia-vtool/blob/master/.github/workflows/python-publish.yml#L104.

This will test the new wheel inside an isolated cibuildwheel environment and will fail to build the wheel if the tests fail.  This also allows us to easily change which OS and python versions to target as cibuildwheel will take care of the complexity of getting an isolated environment ready to check the final wheel.

Reply:
I had suggested we not use `CIBW_TEST_COMMAND` early on, because I wanted to ensure the built code works outside of the build environment (i.e. absence of libraries, etc.).

Reply:
I believe that's the point, but we can build our own testing.  I'd rather trust the utility of `CIBW_TEST_COMMAND` and push a fix or just an issue if we find environment isolation issues.